client/src/rpc_client.rs

```diff
@@ -1146,7 +1147,14 @@ impl RpcClient {
                 signature,
             ));
             sleep(Duration::from_millis(500));
-            confirmations = self.get_num_blocks_since_signature_confirmation(&signature)?;
+            confirmations = self
+                .get_num_blocks_since_signature_confirmation(&signature)
+                .unwrap_or(confirmations);
+            if now.elapsed().as_secs() >= 120 {
```

Comment:
```suggestion
            if now.elapsed().as_secs() >= solana_sdk::clock::MAX_HASH_AGE_IN_SECONDS {
```

Reply:
Perfect, I was going to ask you about that explicitly.